src/test/test_snap_mapper.cc

```diff
@@ -573,6 +587,43 @@ class MapperVerifier {
       }
       hoids.clear();
     }
+
+    return ret;
+  }
+  // must be called with lock held
+  // will keep trimming until reaching max_count or failing a call to __trim_snap()
+  void trim_snap_force(snapid_t snapid, int max_count, vector<hobject_t> & out) {
+    int guard = 1000;
+    vector<hobject_t> tmp;
+    unsigned prev_size = 0;
+    while (tmp.size() < (unsigned)max_count) {
+      unsigned req_size = max_count - tmp.size();
+      __trim_snap(snapid, req_size, tmp);
+      if (prev_size < tmp.size()) {
+	prev_size = tmp.size();
+      }
+      else{
+	// we were unable to find anything to trim -> breakout
```

Comment:
How does `tmp.size() >= (unsigned)max_count` suggest that we were unable to trim?

Reply:
we keep adding objects to the tmp vector while prev_size keeps its size before the call to trim.
If tmp.size is not bigger than its prev_size it means no new objects were added or in other words 
// we were unable to find anything to trim -> breakout

Reply:
Can you add this clarification as a comment?

Reply:
done